src/compiler/crystal/program.cr

```diff
@@ -597,4 +597,61 @@ module Crystal
       io << "<Program>"
     end
   end
+
+  {% if flag?(:compiler_debug) %}
+    def self.debug_log (*args)
+      old_stdout_sync = STDOUT.sync?
+      old_stderr_sync = STDERR.sync?
+      STDOUT.sync = true
+      STDERR.sync = true
+
+      puts *args
+
+      STDOUT.sync = old_stdout_sync
+      STDERR.sync = old_stderr_sync
+      nil
+    end
+
+    def self.debug_log (&block)
+      old_stdout_sync = STDOUT.sync?
+      old_stderr_sync = STDERR.sync?
+      STDOUT.sync = true
+      STDERR.sync = true
+
+      yield
+
+      STDOUT.sync = old_stdout_sync
+      STDERR.sync = old_stderr_sync
+      nil
+    end
+{% else %}
+  def self.debug_log (*args)
+    if ENV["COMPILER_DEBUG"]?
```

Comment:
`if ENV["COMPILER_DEBUG"]? || {{flag?(:compiler_debug)}}` is a neater way to do this

Reply:
Will I was not sure if I can mix macros together so I just implemented this way but definitely it is a proper way for doing it.